src/OrchardCore.Modules/OrchardCore.Lists/Services/ContainerService.cs

```diff
@@ -357,31 +357,30 @@ public async Task<IEnumerable<ContentItem>> QueryContainedItemsAsync(string cont
 
         private IQuery<ContentItem> ContainedItemOptionsFilter(ContainedItemOptions containedItemOptions, IQuery<ContentItem> containedItems)
         {
-            if (!string.IsNullOrEmpty(containedItemOptions?.DisplayText) || containedItemOptions.Status != ContentsStatus.None)
+            if (!string.IsNullOrEmpty(containedItemOptions.DisplayText))
             {
-                if (!string.IsNullOrEmpty(containedItemOptions?.DisplayText))
-                {
-                    containedItems.With<ContentItemIndex>().Where(i => i.DisplayText.Contains(containedItemOptions.DisplayText));
-                }
+                containedItems.With<ContentItemIndex>().Where(i => i.DisplayText.Contains(containedItemOptions.DisplayText));
+            }
 
-                if (containedItemOptions.Status != ContentsStatus.None)
+            if (containedItemOptions.Status != ContentsStatus.None)
+            {
+                switch (containedItemOptions.Status)
                 {
-                    switch (containedItemOptions.Status)
-                    {
-                        case ContentsStatus.Draft:
-                            containedItems?.With<ContentItemIndex>().Where(i => !i.Published);
-                            break;
-                        case ContentsStatus.Published:
-                            containedItems?.With<ContentItemIndex>().Where(i => i.Published);
-                            break;
-                        case ContentsStatus.Owner:
-                            var loggedUser = _hca.HttpContext.User.Identity.Name;
-                            if (loggedUser != null)
-                                containedItems?.With<ContentItemIndex>().Where(i => i.Owner == loggedUser);
-                            break;
-                        default:
-                            throw new NotSupportedException("Unknown status filter.");
-                    }
+                    case ContentsStatus.Draft:
+                        containedItems?.With<ContentItemIndex>().Where(i => !i.Published);
+                        break;
+                    case ContentsStatus.Published:
+                        containedItems?.With<ContentItemIndex>().Where(i => i.Published);
+                        break;
+                    case ContentsStatus.Owner:
+                        string currentUserName = _hca.HttpContext?.User?.Identity?.Name;
+                        if (currentUserName != null)
```

Comment:
Stringet továbbra is `IsNullOrEmpty()`-vel kellene checkolni.

Reply:
oké.